src/k8s/console.ts

```diff
@@ -7,86 +7,82 @@ import * as vscode from 'vscode';
 import { Command } from "../odo";
 import { KubeConfigUtils } from "../util/kubeUtils";
 import { OpenShiftItem } from '../openshift/openshiftItem';
+import { CliExitData } from '../cli';
 
 const k8sConfig = new KubeConfigUtils();
-const clusterUrl = k8sConfig.getCurrentCluster().server;
 const project = (k8sConfig.contexts).find((ctx) => ctx.name === k8sConfig.currentContext).namespace;
 
 export class Console extends OpenShiftItem {
 
-    static async fetchClusterVersion() {
-        const versionInfo = await Console.odo.execute(Command.getclusterVersion(), process.cwd(), false);
-        return versionInfo.error;
-    }
-
     static async fetchOpenshiftConsoleUrl() {
-        if (await Console.fetchClusterVersion() === null) {
-            const routeObj = await Console.odo.execute(Command.getOpenshiftClusterRoute());
-            const spec = JSON.parse(routeObj.stdout).items[0].spec;
-            const consoleUrl = `${spec.port.targetPort}://${spec.host}`;
-            return consoleUrl;
-        } else {
+        try {
+            return await Console.odo.execute(Command.showConsoleUrl());
+        } catch (ignore) {
             const consoleUrl = await Console.odo.execute(Command.showServerUrl());
             return consoleUrl.stdout;
         }
     }
 
-    static async openBuildConfig(context: { id: any; }) {
+    static openShift4ClusterUrl(consoleUrl: string | CliExitData): string {
+        return JSON.parse(consoleUrl['stdout']).data.consoleURL;
+    }
+
+    static async openBuildConfig(context: { name: any; }) {
         let url = '';
         if (!context) {
             vscode.window.showErrorMessage("Cannot load the build config");
             return;
         }
         const consoleUrl = await Console.fetchOpenshiftConsoleUrl();
-        if (await Console.fetchClusterVersion() === null) {
-            url = `${consoleUrl}/k8s/ns/${context.id}/buildconfigs`;
+        if (consoleUrl['stdout']) {
+            url = `${Console.openShift4ClusterUrl(consoleUrl)}/k8s/ns/${context.name}/buildconfigs`;
```

Comment:
This does not work for 4.2.2. It should be
``` JavaScript
url = `${Console.openShift4ClusterUrl(consoleUrl)}/k8s/ns/${project}/buildconfigs/${context.name}`;
```

The same applies to other resources in OpenShift 4.2.2 cluster.